File: src/optic/lua/ssa/instructions/Block.java
package optic.lua.ssa.instructions;

import optic.lua.ssa.*;
import org.jetbrains.annotations.NotNull;

import java.util.*;
import java.util.stream.Stream;

public final class Block implements Step {
	private final List<Step> steps;

	public Block(List<Step> steps) {
		this.steps = List.copyOf(steps);
	}

	@Override
	public @NotNull Stream<Step> children() {
		return steps.stream();
	}

	@Override
	public String toString() {
		return "do";
	}
}

File: src/optic/lua/ssa/instructions/Block.kt
package optic.lua.ssa.instructions

import optic.lua.ssa.Step
import java.util.stream.Stream

class Block(val steps: List<Step>) : Step {
    override fun children(): Stream<Step> {
        return steps.stream()
    }

    override fun toString(): String {
        return "do"
    }
}
